### ISSUE TITLE
Noticed something about Wifi

### ISSUE THREAD

**Issue by test2466** [jakeday/linux-surface#576]
Hello,

I don't know anything about editing the kernel and am fairly new to linux. I am running Qzed's 5.3.1-surface kernel on a Surface Book 1. Suspend is working very well and so far I have had no issues with it.

I am aware of the wifi issues where the wifi device sometimes fails to come up after resuming from suspend. I tried playing with the sleep.sh script by adding "systemctl restart network-manager.service" after the previous NetworkManager restart command. This didn't help as sometimes wifi still failed to come up after suspend. I am sure many are aware of this. 

In order to get around this I have had to run "systemctl restart network-manager.service" in the terminal after suspend. 

I noticed that sometimes for some reason if I run the command "systemctl restart network-manager.service" while the wifi is working it will cause the wifi to suddenly disappear. This leads me to believe that it may be possible that the restart NetowrkManager command in the sleep.sh script could sometimes cause the wifi to quit working as a similar command is present in the script. This is consistent with my observations because after resuming from suspend the wifi usually still works but sometimes it quits working. 

I don't know if this helps anyone but I just wanted to post my observations here in case it is of any help.

**Comment by StollD** [comment]
Are you using the script from this repository, or the updated one from qzed? https://github.com/qzed/linux-surface/blob/v5.3/root/lib/systemd/system-sleep/sleep

I am using the updated one on a Surface Book 2 and the wifi is rock solid after suspending. The issues that you describe sound like what I experienced with the old script.

**Comment by test2466** [comment]
Thank you. I had the old script. I was unaware that I had to replace it manually as I thought that the setup.sh script or just installing the new kernel would automatically add the new scripts. I will go ahead and close this issue.

**Comment by test2466** [comment]
Hi

I just wanted to add a update here. I used the updated sleep.sh qzed script on a Surface Book 1 (GTX 965M) and unfortunately I still had issues with the wifi not being available after suspend. 

I added a workaround to the qzed sleep.sh script on my computer which made wifi stop breaking after suspend (at least as far as I have experienced). I went into the script and added:

"
systemctl restart network-manager.service
systemctl restart network-manager.service

"

after the last line which restarts NetworkManager.


I'm not sure why but for some reason this has made the wifi not break on my SB1 after suspend. It is redundant to add the code twice but sometimes when I restart network manager in the terminal after resume it breaks the wifi if it was working before. I have to run the command a second time for it to start working again without rebooting. For this reason I added the command twice and the wifi has not broken so far after suspend.
